### DISCUSSION TITLE
分享在威联通nas上部署成功。

### DISCUSSION THREAD

**Discussion by jianjam** [oddfar/campus-imaotai#253] [分享部署教程]
qanp设备是264c 。最新系统。
本人也是小白，忙活了一个晚上。在这里简单分享一下过程吧。

整个服务需要大致需要需要4个东西。 mysql，redis，jar（java后台服务端），nginx（dist前台网页端）。

自带的docker compose 一键安装包已经可以傻瓜式部署了。很方便。但是有个弊端是会单独运行一个sql和redis。所以我想使用别的方法。

**数据库：**
qnap的应用中心里自带一个marirdb 的软件。遂安装之。我装的是 marirdb 10 ，安装后需要设置一个密码和端口。

如果要对数据库进行管理需要打开控制台——应用程序——web服务器，启动web服务器。然后再在商店里安装phpmyadmin。安装好后登录 marirdb 10数据库 ，用户名root，密码就是上面设置的。

登录进入后手动创建一个 campus_imaotai的数据库，创建完点击进入后在sql选项卡里运行 数据库初始化文件，就是 博主提供的sql文件，粘贴复制就行。运行完会看到campus_imaotai 里有各种数据表，就说明OK 了。

**redis：**
这里我使用的时docker hub上redis官方的镜像 ，docker pull redis  拉取镜像 。然后使用命令行创建容器：
`docker run --name redis --hostname redis -itd  --network=host  -v /share/docker/redis/data:/data redis redis-server --save 60 1 --loglevel warning`
这里注意-v挂载数据。路径替换成自己的。网络我用的时host模式，如果要使用bridge，注意映射相关端口。把 --network=host 换成 -p  6379:6379 ，前面是你想要redis 使用的对外的端口，后面必须是6379.

**java服务端：**
这个博主也在说明文档里写的很详细了。

https://oddfar.github.io/campus-doc/pages/8f2aa8/#docker%E5%90%AF%E5%8A%A8%E5%90%8E%E7%AB%AF

照着这个里面来就行，注意修改mysql 也就是上面 marirdb 10 的 地址，端口，密码。 redis也修改成你设置的端口。

然后使用运行docker 启动服务后观察日志，如果日志显示正常。访问 nas的地址+8160,就能看到文字提示说明后端服务OK 了。
我这里显示的是v1.0.11，但是github上最新版好像是v1.0.12.可能hub上的镜像是老版本的吧。我还不知道咋搞。还有就是 github上的 campus-imaotai-api 后端镜像和docker hub上那个一样都是后端吗我没试过。

**前端网页：**
前端使用的是作者在github上提供的 campus-imaotai-web 镜像。吐槽一下，拉取这个镜像足足用了我1一个小时。。。。
我是使用qnap 自带的docker 容器管理器创建的。不是用命令行，以为默认端口是80，注意尽量不要使用host模式，用桥接模式，你可以更改一个未使用的端口来。也不需要挂载目录。再设置一个重启策略就可以了。运行后打开nas ip + 你设置的端口就能看到登录页面了。
其实不使用容器，下载作者提供的整个dist文件夹。放到nas上，使用nas自带的虚拟主机服务直接打开html也可以。

题外话：如果你看到404报错，那就需要再改个设置，我就遇到了！！！

按理说这上面几个容器服务都是在nas上运行的。不是host模式就是桥接，应该是说是互通的。但是前端一直连不到后端。然后我用qnap 自带容器管理进到前端容器的命令行 ，找到/etc/nginx/nginx.conf 文件。靠近下面有一行proxy_pass http://localhost:8160/; 就是服务端的地址和端口。用localhost怎么都连不上。使用 vi 我把它改成nas ip + 端口，然后重启一下容器就好了。

OK 大功告成！

在这里想能不能把一些参数放到镜像的环境变量上，比如后端连sql和rdis的ip，端口，账号密码。application-prod.yml 文件就可以直接在镜像里打包了。前端连后端的地址，端口等。俺也是小白。只是提一个建议。希望可以帮助到需要的人。

**Comment by oddfar** [comment]
docker hub的镜像没推（因为现在懒的构建自动化）
GitHub的镜像：campus-imaotai-web 是前端镜像、campus-imaotai-api是后端镜像

拉的慢原因是在国外，可以借助其他人工具
🚀 gcr.io、registry.k8s.io、k8s.gcr.io、quay.io、ghcr.io 等国外镜像加速下载服务：https://github.com/togettoyou/hub-mirror

**Reply by jianjam** [reply]
campus-imaotai-api 是最新的，而且可以无缝替换掉dockerhub上的是吧。OK明白了。

**Comment by Young9229** [comment]
我的server服务运行不了，不停重启，有遇到的小伙伴吗，怎么解决啊
